Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Use the official Golang image to build the Go binary
+FROM golang:1.22 AS builder
+
+# Set the Current Working Directory inside the container
+WORKDIR /app
+
+# Copy go.mod and go.sum files and download dependencies
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy the source code into the container
+COPY . .
+
+# Build the Go app
+RUN GOOS=linux GOARCH=amd64 go build -ldflags="-s" -o main cmd/productivite/main.go
+
+# Use a minimal base image to run the Go binary
+FROM alpine:latest
+
+# Install necessary packages
+RUN apk --no-cache add ca-certificates
+
+# Set the Current Working Directory inside the container
+WORKDIR /root/
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider more secure working directory</b></div>
<div id="fix">

Consider using `/app` instead of `/root/` as the working directory for better security practices. Running from `/root` directory could expose sensitive information if there are any vulnerabilities.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -24,10 +24,12 @@
 -WORKDIR /root/
 +RUN adduser -D appuser
 +WORKDIR /app
 +USER appuser
 -COPY --from=builder /app/main .
 -RUN chmod +x /root/main
 -RUN ls -l /root/main
 +COPY --from=builder /app/main .
 +RUN chmod +x /app/main
 +RUN ls -l /app/main
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/aakashgusain1990/productivite/pull/14#issuecomment-2563464674>#148ccb</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged